src/main.rs

```diff
@@ -567,6 +569,22 @@ fn main() {
     }
 }
 
+async fn open_file_limit() -> usize {
+    tokio::task::spawn_blocking(|| {
+        unsafe { libc::getdtablesize() }
+            .try_into()
+            .expect("getdtablesize(): failed to fetch RLIMIT_NOFILE")
+    })
+    .await
+    .expect("spawn_blocking: failed to get file limit")
+}
+
+async fn get_open_fds_count() -> usize {
+    tokio::task::spawn_blocking(|| std::fs::read_dir("/proc/self/fd").unwrap().count())
+        .await
+        .unwrap()
```

Comment:
There are several unwraps used here, were we'd really want an `expect()` with an explanation.